enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -889,27 +891,31 @@ type copyData struct {
 
 func (mc *MigrationCache) copyDataInBackground() error {
 	rateLimiter := rate.NewLimiter(rate.Limit(mc.maxCopiesPerSec), 1)
-
-	for {
-		if err := rateLimiter.Wait(context.Background()); err != nil {
-			return err
-		}
-
-		select {
-		case <-mc.quitChan:
-			// Drain copy channel on shutdown
-			// We cannot close the channel before draining because there's a chance some cache requests may be
-			// concurrently queued as the cache is shutting down, and if we try to enqueue a copy to the closed
-			// channel it will panic
-			for len(mc.copyChan) > 0 {
-				c := <-mc.copyChan
-				mc.copy(c)
+	eg := &errgroup.Group{}
+	for i := 0; i < mc.numCopyWorkers; i++ {
+		eg.Go(func() error {
+			for {
+				if err := rateLimiter.Wait(context.Background()); err != nil {
+					return err
+				}
+				select {
+				case <-mc.quitChan:
+					// Drain copy channel on shutdown
+					// We cannot close the channel before draining because there's a chance some cache requests may be
+					// concurrently queued as the cache is shutting down, and if we try to enqueue a copy to the closed
+					// channel it will panic
+					for len(mc.copyChan) > 0 {
```

Comment:
This for loop won't work properly if there are multiple goroutines. You can move it out of the worker code.

Reply:
Done.